### ISSUE TITLE
Cannot play with mpv on Linux and cannot set format from text config

### ISSUE THREAD

**Issue by Digitalone1** [streamlink/streamlink#2769]
## Bug Report

- [x] This is a bug report and I have read the contribution guidelines.

### Description
Using streamlink on Arch Linux and I'm not able to play Twitch streams with mpv.
I created a text config inside `~/config/streamlink/config` with only `player=mpv`. It's parsed by command line, but streamlink is not able to start mpv:

```
[cli][info] Available streams: audio_only, 160p (worst), 360p, 480p, 720p, 720p60, 1080p60 (best)
[cli][info] Opening stream: 1080p60 (hls)
[cli][info] Starting player: mpv
error: Failed to start player: mpv (Process exited prematurely)
[cli][info] Closing currently open stream...

```

Tried also with `player=/usr/bin/mpv` but nothing changed.

Besides, I can't find a way to set a preferred format from the text configuration. Is this intended or am I missing something?
`STREAM=1080p60` is not parsed, `stream-sorting-excludes=">1080p"` gives me error, but reading documentation should be correct. Is there a way to specify a format from text config? I can do it with youtube-dl. Something like `-f best[height<=1080]`

In addition, the parameter reported [here](https://streamlink.github.io/issues.html#issues-player-caching) regarding mpv is wrong. As stated [here](https://mpv.io/manual/master/#options-cache), --cache parameter only accept _yes_, _no_ and _auto_.

**Comment by Digitalone1** [comment]
Okay, I solved compiling streamlink-git from AUR. So old version is not working anymore with mpv.

Please, release a new version, so Arch upstream repository will be updated.

Found out `default-stream` option to select a specified format from text config.

I'd like to ask something different: `ringbuffer-size` option will cache the stream to an amount of size before sending it to mpv or mpv will read instantly without waiting? I know I can set a specific cache on mpv, but it's a little tricky and I'd like to set something on streamlink if it's possible, but I'm not sure if it's working as I intended.

I specified `ringbuffer-size=20M`, but mpv is starting very fast, so I assume it is not waiting.

Thanks

**Comment by bastimeyer** [comment]
> Cannot play with mpv on Linux

The issue has just been fixed in #2765 (for the MPV 0.30.1 release). There will be a Streamlink patch release soon. We are aware of this issue. The MPV devs are breaking the semantic versioning scheme with this change btw, as it would usually require a major version bump on their side, even if they called the old parameter format "legacy".

> In addition, the parameter reported [here](https://streamlink.github.io/issues.html#issues-player-caching) regarding mpv is wrong. As stated [here](https://mpv.io/manual/master/#options-cache), --cache parameter only accept _yes_, _no_ and _auto_.

The documented MPV `cache` parameter was changed in MPV's 0.30.0 release not too long ago. I'll open a PR with a docs fix later. Thanks for pointing this out.

> I'd like to ask something different: `ringbuffer-size`

The ringbuffer is Streamlink's internal stream buffer for segmented streams for example. You only need to increase it if you're downloading lots of stream segments in parallel. In regards to HLS streams (eg. on Twitch.tv), this isn't the case if you didn't increase the hls-segment-threads parameter. The player will "consume" the data immediately once a segment gets finished downloading (assuming the correct order), depending on the live-edge value which defines how far off you're from the current live position of the stream.

**Comment by Digitalone1** [comment]
Thanks. So, to prevent buffering, if I set 3 segment-threads and live edge to 10, which is the recommended ringbuffer?

**Comment by bastimeyer** [comment]
I can't tell you, because it depends on the size of the segments, which can vary a lot.
Having a huge live-edge value and number of segment threads doesn't help you with buffering issues if your connection is not good enough to watch streams in a certain bitrate. It can only help if your connection is not reliable enough and you're having drops/spikes during the transmission, eg. on wifi networks, or if you're somehow limited by the individual segment downloads. Increasing the live-edge value essentially gives Streamlink time to catch up after short timeouts without your player noticing it, because it'll still have enough data (in its own buffer and) in Streamlink's buffer to read from.

**Comment by Digitalone1** [comment]
Okay, got it. Thank you.

**Comment by arcooke** [comment]
I'm still getting this on 0.32 when launching a stream from streamlink.  Should I open a new ticket or is it fine to add to this one?

```
[cli][info] Found matching plugin hls for URL http://..../chunks.m3u8
[cli][info] Available streams: live (worst, best)
[cli][info] Opening stream: live (hls)
[cli][info] Starting player: mpv
error: Failed to start player: mpv (Process exited prematurely)
[cli][info] Closing currently open stream...

$ mpv --version
mpv 0.32.0 Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
 built on Fri Jan 31 16:16:35 UTC 2020
ffmpeg library versions:
   libavutil       56.31.100
   libavcodec      58.54.100
   libavformat     58.29.100
   libswscale      5.5.100
   libavfilter     7.57.100
   libswresample   3.5.100
ffmpeg version: 4.2.2

```

**Comment by gravyboat** [comment]
@arcooke What version of Streamlink and Python are you running? We had a release about a week ago that should have addressed this.

**Comment by Digitalone1** [comment]
It's working properly on my Arch Linux with streamlink installed from upstream repository.

**Comment by arcooke** [comment]
> @arcooke What version of Streamlink and Python are you running? We had a release about a week ago that should have addressed this.

Oops, forgot I was in the streamlink repo and not mpv repo. 

streamlink 0.10.0
Python 3.6.9

Solus

**Comment by mrkapqa** [comment]
having this issue
` streamlink -p mpv https://www.youtube.com/watch?v=9Auq9mYxFEE 480p
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/watch?v=9Auq9mYxFEE
[cli][info] Available streams: 144p (worst), 240p, 360p, 480p, 720p, 1080p (best)
[cli][info] Opening stream: 480p (hls)
[cli][info] Starting player: mpv
error: Failed to start player: mpv (Process exited prematurely)
`

however, when playing streaming through mpv only, it would play.

i am using devuan 3.0 based on debian buster with nvidia card and driver.
